src/codegen/codegen_coreneuron_cpp_visitor.cpp

```diff
@@ -2364,7 +2388,11 @@ void CodegenCoreneuronCppVisitor::print_instance_variable_setup() {
             if (var.is_index || var.is_integer) {
                 return "ml->pdata";
             } else if (var.is_vdata) {
-                return "nt->_vdata";
+                if (var.symbol->has_all_properties(NmodlType::random_var)) {
+                    return "(nrnran123_State**)nt->_vdata";
```

Comment:
Is this how we "cast" these days. Or should it be more decorated?

Reply:
I implemented the above suggestion.